web/src/user_groups.ts

```diff
@@ -225,6 +225,16 @@ export function is_user_in_group(user_group_id: number, user_id: number): boolea
     return false;
 }
 
+function get_display_name_for_system_group_option(setting_name: string, name: string): string {
+    if (setting_name === "direct_message_permission_group") {
+        if (name === "Nobody") {
+            return "Direct messages disabled";
+        }
+        return name.replace(" and ", " or ");
```

Comment:
Again, this string manipulation logic is never a thing one can write in an internationalized product. 

I'm also not sure what the `and` => `or` name substitution is about. But I think structurally, we might want to try to avoid needing this function at all; it feels like a lot of complex to display an error message.

Reply:
So I think you should drop these `and` => `or` changes from the PR entirely; the right fix is to change the option label string. Can you post in the #design thread about this and also the error strings from `zerver/lib/exceptions.py`, and let's just brainstorm there what strings to use that don't require working in the value of the settings (too messy, especially if it's a compound group) or this kind of and=>or wording transformation.

Reply:
Started https://chat.zulip.org/#narrow/stream/101-design/topic/restricting.20PMs/near/1850255 to discuss what the spec should be here.